tests/func/non_interactive.rs

```diff
@@ -251,3 +251,31 @@ fn hash_password() {
         .success()
         .stdout(predicates::str::starts_with("SCRAM-SHA-256$"));
 }
+
+#[test]
+fn force_database_error() {
```

Comment:
Could you add a test to `interactive.rs` that fires up a new *interactive* session once force_database_error has been set?

The `edgedb query` case actually already worked, it was connecting interactively that was broken